pkg/config/config.go

```diff
@@ -119,6 +119,8 @@ type Server struct {
 	EnableExperimentalAdmin bool   `def:"false" desc:"whether to enable the experimental admin interface" mapstructure:"enable-experimental-admin"`
 
 	ScrapeConfigs []*scrape.Config `yaml:"scrape-configs" mapstructure:"-"`
+
+	EnableSelfProfiling bool `def:"true" desc:"enable profiling of pyroscope itself" mapstructure:"enable-self-profiling"`
```

Comment:
This is a relatively minor, but what do you think about doing the opposite, e.g `DisableSelfProfiling`, and making it `false` by default?

The advantage is that when you're running it from command line you can just do `pyroscope --disable-self-profiling`. Otherwise you'd have to do something like `pyroscope --enable-self-profiling false` I think, which is slightly more wordy and less common.

Reply:
This is slightly separate but I'm also thinking maybe we should come up with some unified way of naming booleans. Right now we have multiple different styles, e.g some booleans start with `no-`, others start with `disable-`. I would probably choose to go with `no-` prefix, because then if you need to have a variable that's negative by default you can just name it without a prefix, e.g `--enable-experimental-admin` becomes just `-experimental-admin`. 

/cc @abeaumont @kolesnikovae  @Rperry2174 what do you all think about this?

Reply:
These are the reasons I used `--enable`:

a) for better or for worse it's a boolean, and IMO booleans are better (in most cases) in the affirmative position. do you prefer `isDisabled` or `isEnabled`? what about `!isDisabled`? to me that becomes slightly confusing. then you can also have that classic double negative case of `!isNotPresent`, "we don't need no education" vibes

b) it's a toggle, a switch. It makes it easier to think that you are turning on that feature, even the UIs work better that way https://cdn.pulse2.com/cdn/2020/01/LaunchDarkly.jpg

c) RE `--enable-self-profiling false` vs `--disable-self-profiling` I don't think the brevity influences that much, since it's such an uncommon flag.

d) The point of feature flags is to add them early and them just flip them over. They are not attached to the usage patterns. With this proposal (some things are --disable and some are --enable), w we also have to think "what the default state is going to be" to come up with the name. 

e) Standardization. Goes a bit with point d) but I prefer to have everything following the same pattern.


I am sure I may be able to find even papers on this subject, but my experience says booleans should be used on the affirmative case:
https://www.google.com/search?q=booleans+afirmative

https://softwareengineering.stackexchange.com/questions/196830/boolean-method-naming-affirmative-vs-negative
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members?redirectedfrom=MSDN#names-of-properties
https://stackoverflow.com/questions/1445836/positive-or-negative-boolean-field-names
https://google.github.io/styleguide/jsguide.html#naming-method-names

Reply:
I prefer short flags, vote for `no-` and omitting `enable-`.

Reply:
a,c,d) I agree with you that generally when programming affirmative names are better and it's a good practice. But in this particular context (command line interface) it seems to be a common practice for boolean flags to be inverted with --no prefix when the default is positive.

b) I agree that for some flags adding `enable-` prefix actually improves the name because it adds extra context making it clear that this is a boolean flag.

I still think `no-self-profiling` would be a better name here.

---

We're also mixing this conversation with the one about feature flags. I could see how for feature flags in particular a whole different approach might be better. I think that should be a separate converstaion.

Reply:
Ok will change it to `--no`.

>it seems to be a common practice for boolean flags to be inverted with --no prefix when the default is positive.
and when we figure out that users actually don't like having self profiling turned on and we decide to make it off by default?